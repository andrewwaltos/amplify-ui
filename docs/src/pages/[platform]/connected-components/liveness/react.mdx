## FaceLivenessDetector

FIXME fill this out
