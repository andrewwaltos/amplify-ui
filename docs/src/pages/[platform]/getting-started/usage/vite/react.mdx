import { Tabs, TabItem } from '@aws-amplify/ui-react';
import { Example, ExampleCode } from '@/components/Example';
import { BasicShoppingCard, AdvancedShoppingCard } from '../shared';
import { InstallScripts, TerminalCommand } from '@/components/InstallScripts.tsx';

## Tutorial

In this brief tutorial, we're going to build a basic shopping card component using [Vite.js](https://vitejs.dev/) and Amplify UI. Then, we'll elaborate on our shopping card to play with the possibilities of using Amplify UI.

### Setup and Installation

First, execute the command below in your terminal. When prompted for the name of your project, enter `amplify-ui-demo`.  Source: [Vite Guide](https://vitejs.dev/guide/)
<Tabs>
  <TabItem title="JavaScript">
    npm 6.x
    <TerminalCommand command="npm create vite@latest amplify-ui-demo --template react && cd amplify-ui-demo" />
    npm 7+, extra double-dash is needed:
    <TerminalCommand command="npm create vite@latest amplify-ui-demo -- --template react && cd amplify-ui-demo"/>
  </TabItem>
  <TabItem title="TypeScript">
    #### npm 6.x
    <TerminalCommand command="npm create vite@latest amplify-ui-demo --template react-ts && cd amplify-ui-demo
    " />
    #### npm 7+, extra double-dash is needed:
    <TerminalCommand command="npm create vite@latest amplify-ui-demo -- --template react-ts && cd amplify-ui-demo"/>
  </TabItem>
</Tabs>
<p>

Next, install the Amplify UI React package:
</p>
<TerminalCommand command="npm install @aws-amplify/ui-react aws-amplify"/>


When working with a [Vite](https://vitejs.dev) project you must make a few modifications to polyfill Node globals. Please follow the steps below:

**1.** Add the following script tag to the `index.html` file right before the `</body>` tag. This will only run on the client side and will polyfill Node globals.

```html
  <script>
    window.global = window;
    window.process = {
      env: { DEBUG: undefined },
    };
    var exports = {};
  </script>
</body>
```

**2.** Update the `vite.config.js` and add a resolve alias inside the `defineConfig({})` as seen below.

```js
export default defineConfig({
  plugins: [react()],
  resolve: {
      alias: [
      {
        find: './runtimeConfig',
        replacement: './runtimeConfig.browser',
      },
    ]
  }
})
```

### Basic Demo

Open up the project in your IDE and delete the contents of the `index.css` file. 

Next replace the contents of the `App.jsx` or `App.tsx` file with the code below:

<Example>
  <ExampleCode>
  
```tsx file=./examples/BasicExampleCode.tsx

```

  </ExampleCode>
  <BasicShoppingCard />
</Example>

When you start the app (`npm run dev`), you should see a basic shopping card component.

### Advanced Demo

If youâ€™d like to play around with a more comprehensive demo of the Amplify UI library, create a separate file under the `pages` folder named `paintings.js`, and copy/paste this code into it:

<ExampleCode>

```tsx file=./examples/paintings.ts

```

  </ExampleCode>

Then, replace the contents of `App.js` or `App.tsx` with the code below, restart the app, and experience all the beautiful components in action!

<Example>
  <ExampleCode>

```tsx file=./examples/AdvancedExampleCode.tsx

```

  </ExampleCode>
  <AdvancedShoppingCard />
</Example>
