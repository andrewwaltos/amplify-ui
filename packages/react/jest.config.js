module.exports = {
  // TODO: change this back when we add unit tests to StorageManager
  collectCoverage: false,
  collectCoverageFrom: ['<rootDir>/src/**/*.(ts|tsx)'],
  // ignore coverage for top level "export", PrimitiveCatalog, and style files
  coveragePathIgnorePatterns: [
    '<rootDir>/src/(index|internal|PrimitiveCatalog|styles).(ts|tsx)',
  ],
  coverageThreshold: {
    global: {
      branches: 80,
      functions: 70,
      lines: 91,
      statements: 91,
    },
  },
  globals: { 'ts-jest': { tsconfig: 'tsconfig.json' } },
  moduleNameMapper: {
    '^react$': '<rootDir>/node_modules/react',
  },
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  preset: 'ts-jest',
  setupFilesAfterEnv: ['./jest.setup.ts'],
  testEnvironment: 'jsdom',
};
