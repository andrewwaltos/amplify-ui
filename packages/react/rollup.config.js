// rollup.config.js
import { defineConfig } from 'rollup';
import typescript from '@rollup/plugin-typescript';
import { terser } from 'rollup-plugin-terser';
import styles from 'rollup-plugin-styles';
import externals from 'rollup-plugin-node-externals';

const input = [
  'src/index.tsx',
  'src/internal.tsx',
  'src/geo.ts',
  'src/models.ts',
];

const config = defineConfig([
  // CJS config
  {
    input,
    output: {
      dir: 'dist',
      format: 'cjs',
      sourcemap: false,
    },
    plugins: [
      externals({ include: /^@aws-amplify/ }),
      typescript({ declarationDir: 'dist/types', sourceMap: false }),
      terser(),
    ],
  },
  // ESM config
  {
    input,
    output: {
      dir: 'dist/esm',
      format: 'es',
      entryFileNames: '[name].mjs',
      preserveModules: true,
      preserveModulesRoot: 'src',
      sourcemap: false,
    },
    plugins: [
      externals({ include: /^@aws-amplify/ }),
      typescript({ outDir: 'dist/esm', declaration: false, sourceMap: false }),
      terser(),
    ],
  },
  // CSS config
  {
    input: ['src/styles.ts', 'src/geoStyles.ts'],
    output: {
      dir: 'dist',
      format: 'cjs',
      assetFileNames: '[name][extname]',
    },
    plugins: [styles({ mode: ['extract'] })],
  },
]);

export default config;
