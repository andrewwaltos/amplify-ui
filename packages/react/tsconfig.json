{
  "compilerOptions": {
    "baseUrl": ".",
    "declaration": true,
    "esModuleInterop": true,
    "importHelpers": true,
    "jsx": "react",
    "module": "esnext",
    "moduleResolution": "node",
    "outDir": "dist/esm",
    "skipLibCheck": true,
    "sourceMap": true,
    "rootDir": "./src",
    "target": "es2015"
  },
  "exclude": ["node_modules", "src/**/*.test.ts", "src/**/*.test.tsx"],
  "include": ["src/**/*.ts", "src/**/*.tsx"]
}
