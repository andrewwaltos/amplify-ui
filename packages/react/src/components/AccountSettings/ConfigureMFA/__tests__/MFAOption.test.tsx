import React from 'react';

import { render, screen } from '@testing-library/react';
import MFAOption from '../MFAOption';

describe('MFAOption', () => {
  it('renders as expected when mfaType is sms', async () => {
    const { container } = render(<MFAOption mfaType="sms" />);

    const smsOption = await screen.findByRole('radio', { name: /Use SMS/ });
    expect(smsOption).toHaveAttribute('value', 'sms');

    expect(container).toMatchSnapshot();
  });

  it('renders as expected when mfaType is totp', async () => {
    const { container } = render(<MFAOption mfaType="totp" />);

    const smsOption = await screen.findByRole('radio', { name: /Use an app/ });
    expect(smsOption).toHaveAttribute('value', 'totp');

    expect(container).toMatchSnapshot();
  });
});
