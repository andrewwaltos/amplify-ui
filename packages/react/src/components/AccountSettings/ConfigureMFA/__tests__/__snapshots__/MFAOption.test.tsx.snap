// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MFAOption renders as expected when mfaType is sms 1`] = `
<div>
  <label
    class="amplify-flex amplify-radio"
  >
    <span
      class="amplify-text amplify-radio__label"
    >
      <p
        class="amplify-text"
        style="font-weight: var(--amplify-font-weights-bold);"
      >
        Use SMS
      </p>
      <p
        class="amplify-text"
      >
        Receive an SMS on your phone with a code
      </p>
    </span>
    <input
      aria-invalid="false"
      class="amplify-input amplify-field-group__control amplify-visually-hidden amplify-radio__input"
      name="default"
      type="radio"
      value="sms"
    />
    <span
      aria-hidden="true"
      class="amplify-flex amplify-radio__button"
    />
  </label>
</div>
`;

exports[`MFAOption renders as expected when mfaType is totp 1`] = `
<div>
  <label
    class="amplify-flex amplify-radio"
  >
    <span
      class="amplify-text amplify-radio__label"
    >
      <p
        class="amplify-text"
        style="font-weight: var(--amplify-font-weights-bold);"
      >
        Use an app
      </p>
      <p
        class="amplify-text"
      >
        Use an application on your phone. Applications include cloud-based TOTP (temporary one time password) apps such as Authy, Google Authenticator, and 1password.
      </p>
    </span>
    <input
      aria-invalid="false"
      class="amplify-input amplify-field-group__control amplify-visually-hidden amplify-radio__input"
      name="default"
      type="radio"
      value="totp"
    />
    <span
      aria-hidden="true"
      class="amplify-flex amplify-radio__button"
    />
  </label>
</div>
`;
